autoload/zsh-learn-Savel

```diff
@@ -1,19 +1,42 @@
 # -*- mode: sh -*-
 # vim: set ft=sh:
 function zsh-learn-Savel(){
-
-    if [[ -z "$1" ]]; then
-        echo "usage: zsh-learn-Savel <learning> [<category>]" >&2
-        return 1
-    fi
+    function usage() {
+        echo "Usage: zsh-learn-Savel [options] <learning>\nOptions:\n -e    Edit learning in default editor\n -c    Set a custom category"
+    }
 
     local category learning
-
     category="programming"
-    learning="$(printf -- '%s' "$1" | sed 's@^[[:space:]]*@@;s@[[:space:]]*$@@')"
 
-    if [[ -n "$2" ]]; then
-        category="$2"
+    while getopts "ec:" opt; do
+        case $opt in
+
+        e)
+            filename="/tmp/edit-learning"
+            touch $filename
+            $EDITOR $filename
+            learning=$(cat $filename)
+            rm $filename
+            ;;
+        c)
+            category=$OPTARG
```

Comment:
I moved the category to a `-c` option instead of arg 2. This makes it compatible when using -e for editor when you don't pass a first arg for learning. Let me know if this breaks anything you have using this.

Reply:
I would use `$TMPDIR` over `tmp` like `${TMPDIR:-/tmp}`.

Reply:
`-c` option is fine.  options over positional parms.

Reply:
also change `$EDITOR` to `${=EDITOR}` for word splitting purposes